src/libs/Network/processRequest.js

```diff
@@ -0,0 +1,19 @@
+import HttpUtils from '../HttpUtils';
+import enhanceParameters from './enhanceParameters';
+import * as NetworkEvents from './NetworkEvents';
+
+/**
+ * @param {Object} request
+ * @param {String} request.command
+ * @param {Object} request.data
+ * @param {String} request.type
+ * @param {Boolean} request.shouldUseSecure
+ * @returns {Promise}
+ */
+export default function processRequest(request) {
+    const finalParameters = enhanceParameters(request.command, request.data);
+    const cancelRequestTimeoutTimer = NetworkEvents.startRequestTimeoutTimer();
```

Comment:
Sorry, what happens if the request times out?

Reply:
We "trigger a connection recheck", but I'll leave a comment

Reply:
I kinda move this so we wouldn't have to explain what happens and just emit and event. But the event is hardcoded in :D